### ISSUE TITLE
[Events] Failed to generate an user event

### ISSUE THREAD

**Issue by claudiuolteanu** [asterisk/ari-py#10]
Hello folks,

When I generate an user event I receive the following error:
"_AttributeError: 'dict' object has no attribute 'paramType'_" , raised on **swaggerpy/client.py line 73**. (more details here [1]). 

I believe that this is a _swagger-py_ limitation since the parameter type is expected to be '_path_' or '_query_' and when you generate an user event the type of '_variables_' parameter will be '_body_' .

Below you can find the code which generates the error:

``` python
connection = ari.connect('http://localhost:8088', 'user', 'password') # connect to the ARI server
connection.events.userEvent(eventName="NewEvent", 
                            application='Test', 
                            variables={"variables" : {"var1":"value1"}}) # generate the user event
```

[1] 
![error](https://cloud.githubusercontent.com/assets/1594079/4268995/b71da2f0-3cb6-11e4-8406-c7f9da4724e6.jpg)

All the best,
Claudiu

**Comment by matt-jordan** [comment]
You're correct. Right now, swagger-py does not support body parameters - as such, it doesn't have support for a dictionary of values that should be converted to JSON.

**Comment by matt-jordan** [comment]
And, this is a problem in swagger-py, not ari-py (it just happens to break ari-py as well).

**Comment by litnimax** [comment]
Is swagger-py expected to support body params?
Who is responsible for developing it?
Thanks.

**Comment by matt-jordan** [comment]
No one is "responsible". It's an open source project, currently under the Digium repo (since that's what we had available at the time).

https://github.com/digium/swagger-py

Anyone can make a pull request to it.

**Comment by nhorelik** [comment]
Note that if you really need this functionality, you can still simply make posts manually.  For example:

``` python
import json
import requests
data = json.dumps({'variables': {'myvar': 'myval'}})
headers = {'Content-type': 'application/json', 'Accept': 'application/json'}
auth = ('username', 'password')
params = {'eventName': 'myevent', 'application': 'myappname'}
requests.post('http://127.0.0.1:8088/ari/events/user/myevent',data=data, headers=headers, auth=auth, params=params)
```

I'd prefer to use ari-py everywhere though, so hopefully this functionality will be added in the next release!  See digium/swagger-py#9.
